pkg/common/cns-lib/volume/manager.go

```diff
@@ -1530,3 +1534,134 @@ func (m *defaultManager) QuerySnapshots(ctx context.Context, snapshotQueryFilter
+		log.Errorf("unable to find the task result for CreateSnapshots task from vCenter %q. taskID: %q, opId: %q createResults: %+v",
+			m.virtualCenter.Config.Host, createSnapshotsTaskInfo.Task.Value, createSnapshotsTaskInfo.ActivationId, createSnapshotsTaskResult)
+		return "", "", "", nil, err
+	}
+
+	// Handle snapshot operation result
+	createSnapshotsOperationRes := createSnapshotsTaskResult.GetCnsVolumeOperationResult()
+	if createSnapshotsOperationRes.Fault != nil {
+		return "", "", "", nil, logger.LogNewErrorf(log,
+			"failed to create snapshots for cns volume %s. fault: %q, opId: %q",
+			volumeID, spew.Sdump(createSnapshotsOperationRes.Fault), createSnapshotsTaskInfo.ActivationId)
+	}
+
+	snapshotCreateResult := interface{}(createSnapshotsTaskResult).(*cnstypes.CnsSnapshotCreateResult)
+	snapshotID := snapshotCreateResult.Snapshot.SnapshotId.Id
+	snapshotVolumeID := snapshotCreateResult.Snapshot.VolumeId.Id
+	snapshotDescription := snapshotCreateResult.Snapshot.Description
+	snapshotCreateTime := snapshotCreateResult.Snapshot.CreateTime
+
+	log.Infof("CreateSnapshot: Snapshot created successfully. SnapshotID: %q, SnapshotCreateTime: %q, opId: %q", snapshotID, snapshotCreateTime, createSnapshotsTaskInfo.ActivationId)
+
+	return snapshotID, snapshotVolumeID, snapshotDescription, &snapshotCreateTime, err
+}
+
+func (m *defaultManager) DeleteSnapshot(ctx context.Context, volumeID string, snapshotID string) error {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectCns(ctx)
+	if err != nil {
+		log.Errorf("ConnectCns failed with err: %+v", err)
+		return err
+	}
+
+	// Construct the CNS SnapshotDeleteSpec list
+	var cnsSnapshotDeleteSpecList []cnstypes.CnsSnapshotDeleteSpec
+	cnsSnapshotDeleteSpec := cnstypes.CnsSnapshotDeleteSpec{
+		VolumeId: cnstypes.CnsVolumeId{
+			Id: volumeID,
+		},
+		SnapshotId: cnstypes.CnsSnapshotId{
+			Id: snapshotID,
+		},
+	}
+	cnsSnapshotDeleteSpecList = append(cnsSnapshotDeleteSpecList, cnsSnapshotDeleteSpec)
+
+	// Call the CNS DeleteSnapshots
+	log.Infof("Calling CnsClient.DeleteSnapshots: VolumeID [%q] SnapshotID [%q] cnsSnapshotDeleteSpecList [%#v]", volumeID, snapshotID, cnsSnapshotDeleteSpec)
+	deleteSnapshotsTask, err := m.virtualCenter.CnsClient.DeleteSnapshots(ctx, cnsSnapshotDeleteSpecList)
+	if err != nil {
+		log.Errorf("CNS DeleteSnapshots failed from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+
+	// Get the taskInfo
+	deleteSnapshotsTaskInfo, err := cns.GetTaskInfo(ctx, deleteSnapshotsTask)
+	if err != nil || deleteSnapshotsTaskInfo == nil {
+		log.Errorf("Failed to get taskInfo for DeleteSnapshots task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+	log.Infof("DeleteSnapshots: VolumeID: %q, SnapshotID: %q, opId: %q", volumeID, snapshotID, deleteSnapshotsTaskInfo.ActivationId)
```

Comment:
I see some value is logging the actual vc task-id here, that way failures can easily be co-related between csi logs and vc logs.

Reply:
I don't think it is necessary. the taskinfo activationId would always be the prefix of OpID in CSI logs and VC logs. It should be good enough.

Reply:
Sure, OpId is useful as well, but logging the task-id lets you look at the task from vc UI as well. I think it adds value in terms of debugability.

Reply:
task-id is anyway available in CNS log. If it would be would be useful, it can be searched by OpId. Meanwhile, we can also query task-id by activation id in the vpx_task table in VCDB, which is data source of task entries in VC UI.